orchestra/utils/notifications.py

```diff
@@ -83,10 +84,32 @@ def notify_status_change(task, previous_status=None):
                                assignment.worker.user.email]
         }
 
+    if current_worker is None:
+        _notify_slack(task, current_worker)
+        _notify_email(task, message_info)
+
+    else:
+        # TODO(joshblum): Find a good abstraction for checking preferences in
+        # general
+        comm_type = (CommunicationPreference.CommunicationType
+                     .TASK_STATUS_CHANGE.value)
+        comm_pref = CommunicationPreference.objects.get(
+            worker=current_worker,
+            communication_type=comm_type
+        )
+        if comm_pref.can_slack():
+            _notify_slack(task, current_worker)
+        if comm_pref.can_email():
+            _notify_email(task, message_info)
+
+
+def _notify_slack(task, current_worker):
     _notify_internal_slack_status_change(task, current_worker)
     if task.project.slack_group_id:
         _notify_experts_slack_status_change(task, current_worker)
 
+
+def _notify_email(task, message_info):
```

Comment:
we should disable email notifications in here, not in the function above---it's a per-recipient decision